stdlib/public/core/Diffing.swift

```diff
@@ -76,34 +79,44 @@ extension RangeReplaceableCollection {
       into target: inout Self,
       contentsOf source: Self,
       from index: inout Self.Index, count: Int
-    ) {
+    ) throws {
       let start = index
-      source.formIndex(&index, offsetBy: count)
+      if !source.formIndex(&index, offsetBy: count, limitedBy: source.endIndex) {
+        throw _ApplicationError.failed
+      }
       target.append(contentsOf: source[start..<index])
     }
 
-    difference._fastEnumeratedApply { change in
```

Comment:
This hunk is all whitespace changes with the exception of the addition of a `do { … } catch { … }` block and associated sprinklings of `try`.